addons/web/static/src/core/utils/objects.js

```diff
@@ -37,7 +37,7 @@ export const deepEqual = (obj1, obj2) => shallowEqual(obj1, obj2, deepEqual);
  * @return {T}
  */
 export function deepCopy(obj) {
-    return JSON.parse(JSON.stringify(obj));
+    return obj && JSON.parse(JSON.stringify(obj));
```

Comment:
```js
* @param {T} obj An object that is fully JSON stringifiable
```
I don't think I agree with your change. Why would you call it with undefined? Simply check what you have, and call this function if necessary...

Reply:
Damn I didn't read your commit message first... Didn't see it was a kind of a backport. I still disagree with the change though, but that's a question for master now.

Reply:
cc @jum-odoo

Reply:
I discussed this with jum and it was his suggestion  ^^although I fix the exact spot where it failed. I don't mind :)

Reply:
I don't really mind about 17. My concern is about master, and the thing is already done there. I'll check with jum, but I think it's a mistake.

So I won't block this PR as it's a backport